zio-k8s-operator/src/main/scala/com/coralogix/zio/k8s/operator/leader/package.scala

```diff
@@ -19,28 +19,28 @@ package object leader {
       /** Runs the given effect by applying the leader election algorithm, with the guarantee that
         * the inner effect will only run at once in the Kubernetes cluster.
         *
-        * If you want to manage the lock as a ZManaged use [[lease]]
+        * If you want to manage the lock as Scoped use [[lease]]
         *
         * @param f
         *   Inner effect to protect
         */
-      def runAsLeader[R, E, A](f: ZIO[R, E, A]): ZIO[R with Clock, E, Option[A]] =
+      def runAsLeader[R, E, A](f: ZIO[R, E, A]): ZIO[R with Scope, E, Option[A]] =
         lease
-          .use(_ => f.mapBoth(ApplicationError.apply, Some.apply))
+          .zipRight(f.mapBoth(ApplicationError.apply, Some.apply))
           .catchAll((failure: LeaderElectionFailure[E]) =>
             logLeaderElectionFailure(failure).as(None)
           )
 
       /** Creates a managed lock implementing the leader election algorithm
         */
-      def lease: ZManaged[Clock, LeaderElectionFailure[Nothing], Unit]
+      def lease: ZIO[Scope, LeaderElectionFailure[Nothing], Unit]
     }
 
     class Live(contextInfo: ContextInfo.Service, lock: LeaderLock) extends Service {
-      override def lease: ZManaged[Clock, LeaderElectionFailure[Nothing], Unit] =
+      override def lease: ZIO[Scope, LeaderElectionFailure[Nothing], Unit] =
```

Comment:
Same things apply to this as mentioned earlier, about the use of `ZIO.scoped`

Reply:
done.